session/grpc.go

```diff
@@ -44,6 +45,8 @@ func grpcClientConn(ctx context.Context, conn net.Conn) (context.Context, *grpc.
 	dialOpts := []grpc.DialOption{
 		dialer,
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(defaults.DefaultMaxRecvMsgSize)),
```

Comment:
Should this be slightly smaller than the control api size? I'm not sure how much padding there is of messages (since we tunnel the session through the control api), but in this case there's now a scenario in which we could send a valid session api message that's too large for a control api message.

Reply:
This limit isn't something that is used for chunking messages. It is the upper bound that should be unreachable. If something misbehaves and we hit this limit then the only difference is if the error comes from the session API or control API layer.

Reply:
> the only difference is if the error comes from the session API or control API layer.

Fair enough - I think this is what I was getting at before, but agreed, I don't think this matters hugely.